client/src/lib/solana/sol-data.ts

```diff
@@ -189,42 +248,52 @@ export class SolData extends Assignable {
     return VERSION;
   }
 
-  static parseDIDReferenceArray(
+  static async parseDIDReferenceArray(
     dids: (string | DIDVerificationMethod)[] | undefined
-  ): string[] {
-    return DecentralizedIdentifier.parseMaybeArray(dids).map(
+  ): Promise<(string | undefined)[]> {
+    return (await DecentralizedIdentifier.parseMaybeArray(dids)).map(
       (did) => did.urlField
     );
   }
 
-  static parse(document: Partial<DIDDocument> | undefined): SolData {
-    if (document) {
-      return new SolData({
-        version: SolData.parseVersion(document['@context']),
-        did: document.id
-          ? DecentralizedIdentifier.parse(document.id)
-          : DecentralizedIdentifier.empty(),
-        verificationMethod: document.verificationMethod
-          ? document.verificationMethod.map((v) => VerificationMethod.parse(v))
-          : [],
-        authentication: SolData.parseDIDReferenceArray(document.authentication),
-        capabilityInvocation: SolData.parseDIDReferenceArray(
-          document.capabilityInvocation
-        ),
-        capabilityDelegation: SolData.parseDIDReferenceArray(
-          document.capabilityDelegation
-        ),
-        keyAgreement: SolData.parseDIDReferenceArray(document.keyAgreement),
-        assertionMethod: SolData.parseDIDReferenceArray(
-          document.assertionMethod
-        ),
-        service: document.service
-          ? document.service.map((v) => ServiceEndpoint.parse(v))
-          : [],
-      });
-    } else {
-      return SolData.empty();
-    }
+  static async parse(document: Partial<DIDDocument>): Promise<SolData> {
+    const did = document.id
+      ? await DecentralizedIdentifier.parse(document.id)
+      : DecentralizedIdentifier.empty();
+
+    return new SolData({
+      account: did.pdaPubkey,
+      authority: did.authorityPubkey,
+      cluster: did.clusterType,
+      version: SolData.parseVersion(document['@context']),
+      verificationMethod: document.verificationMethod
+        ? await Promise.all(
+            document.verificationMethod.map(
+              async (v) => await VerificationMethod.parse(v)
+            )
+          )
+        : [],
+      authentication: (
+        await SolData.parseDIDReferenceArray(document.authentication)
+      ).filter((val): val is string => val !== undefined),
```

Comment:
(val): val is string => val !== undefined

duplicated everywhere (see above comment about unfriendly return values)

Reply:
Fixed with the above